### ISSUE TITLE
Google Docs Lags

### ISSUE THREAD

**Issue by dalisoft** [AdguardTeam/AdguardForMac#1065]
### Steps to reproduce

1. Enable AdGuard with filters
2. Try to work with Google Docs

### Expected behavior

0-20 seconds

### Actual behavior

After 20 seconds in video

<details><summary>Video:</summary>

https://user-images.githubusercontent.com/3511344/166162290-09e38e68-deab-4482-9f8f-1c057bfa6906.mp4

</details>

### Your environment

* Environment name and version: Edge 101, Firefox 99
* Any specific potentially conflicting software installed: only NextDNS, but that's only for DNS traffic

**Comment by Chinaski1** [comment]
Hello there!

- Please specify whether the DNS protection module is enabled in AdGuard?
- Have you tried disabling all AdGuard filters and reproducing the issue again?

But in any case to troubleshoot this issue, we need to get the app logs.

Here's what we need you to do:

1. Open AdGuard VPN settings --> General --> Advanced settings --> Logging level --> **Record everything**;
2. Reproduce the issue and remeber the exact time it happened;
3. Open Settings --> General --> **Export logs**;
4. Send the archive to `apple@adguard.com` and mention this issue number in the subject.

**Comment by dalisoft** [comment]
@Chinaski1 Yep, seems issue at **Stealth mode** and **AdGuard Tracking Filter**. If i am disable both of them, issue disappears as mentioned above.

**Comment by dalisoft** [comment]
It's weird but uBlock Origin which uses AdGuard filters, works fine

**Comment by dalisoft** [comment]
https://user-images.githubusercontent.com/3511344/167597181-60606c95-bf39-46e8-8fbb-cddbbe239e05.mp4

@Chinaski1 i am debugger and tried disable all filters and check by enabling per filter so i found bug. It's Dandelion's Annoyances List.
Same list i am loaded to uBlock Origin but bug did not occur, i don't know why it happens in AdGuard. hope in fix

**Comment by maxikuzmin** [comment]
@dalisoft I repeated the same steps with the "Adguard Base filter" enabled and  "Dandelion Sprouts's Annoyances List". Result don't repeat.
I think the "Dandelion Sprouts's Annoyances List" filter conflicts with one of the other filters you have. 
I recommend not including filters in huge amount, or you can add Google tables to the Allowlist

**Comment by slavaleleka** [comment]
@AdguardTeam/filters-maintainers could you please take a look is it still reproducible?

**Comment by dalisoft** [comment]
I solved by moving to uBlock Origin for MS Edge and Firefox, currently for Safari using allowlist

**Comment by Alex-302** [comment]
@slavaleleka 

> still

Never noticed this.

**Comment by Chinaski1** [comment]
Since the issue was not on the application side, I am closing this task

**Comment by dalisoft** [comment]
@Chinaski1 @Alex-302 @slavaleleka Thank you for your efforts and trying fix issue. I am re-tried now and yes, issue still exists. Same filter (with same URL) works fine in uBlock Origin in Edge/Firefox but in AdGuard laggy/delays for writing text and choosing cells.
Tried with disabled cache too

https://user-images.githubusercontent.com/3511344/168605735-8982cd23-0eff-4dc7-93eb-a0a32c108082.mp4

I am myself developer and worked with testers too together, i know how hard is reproduce issue but i am attaching video again with disabled cache and twice disabled/enabled the filter which caused laggy (**Dandelion's Annoyances List**. bug not in filter itself), see and let me know it's correct or not.

When disabled **Dandelion's Annoyances List** filter by browsers works/feels again snappy and can use with please. Adding to **Allowlist** is not way because we care about privacy and so using **AdGuard**

**Comment by Alex-302** [comment]
@dalisoft Please try `@@||docs.google.com^$elemhide`

**Comment by dalisoft** [comment]
@Alex-302 You cannot believe but it works. Now Google Sheets snappy as always. Works fast as uBO but for all browsers, thank you billion times.
I have question, it is does not affect to my privacy like Stealth mode or any IP hiding privacy?

**Comment by zloyden** [comment]
> I have question, it is does not affect to my privacy like Stealth mode or any IP hiding privacy?

This rule does not affect to Stealth mode.

**Comment by dalisoft** [comment]
@zloyden Thank you. Now again feeling best of best experience which always has been before
